nncf/torch/quantization/algo.py

```diff
@@ -1025,13 +1026,21 @@ def is_weights(ip: PTTargetPoint) -> bool:
         compression_lr_multiplier = self._get_compression_lr_multiplier()
 
         half_range = False
-        if self.hw_config and not self._disable_saturation_fix and is_weights(primary_ip):
+        if self.hw_config and is_weights(primary_ip):
             if self.hw_config.target_device in ['CPU', 'ANY'] and qconfig.num_bits == 8:
-                nncf_logger.warning('The saturation issue fix will be applied. '
-                                    'Now all weight quantizers will effectively use only 7 bits out of 8 bits. '
-                                    'This resolves the saturation issue problem on AVX2 and AVX-512 machines. '
-                                    'Please take a look at the documentation for a detailed information.')
-                half_range = True
+                if self._saturation_fix == 'enable':
+                    half_range = True
+                    nncf_logger.warning('The saturation issue fix will be applied. '
+                                        'Now all weight quantizers will effectively use only 7 bits out of 8 bits. '
+                                        'This resolves the saturation issue problem on AVX2 and AVX-512 machines. '
+                                        'Please take a look at the documentation for a detailed information.')
+                elif self._saturation_fix == 'enable_for_first_conv_layer':
+                    if target_node in get_first_nodes_of_type(target_model_graph, ['conv2d']):
+                        half_range = True
+                        nncf_logger.warning('The saturation issue fix will be applied. Now first convolution '
+                                            'weight quantizers will effectively use only 7 bits out of 8 bits. '
+                                            'This resolves the saturation issue problem on AVX2 and AVX-512 machines. '
+                                            'Please take a look at the documentation for a detailed information.')
```

Comment:
The texts are duplicated here and above. Could you extract the majority of text as a constant string and format it accordingly in-place?

Reply:
Fixed, thanks!